cmd/bosun/sched/bolt.go

```diff
@@ -103,6 +95,7 @@ func (s *Schedule) save() {
 		log.Printf("wrote %s: %v", name, conf.ByteSize(cw.written))
 		collect.Put("statefile.size", opentsdb.TagSet{"object": name}, cw.written)
 	}
+	s.Unlock()
```

Comment:
What about the return above... do we need to defer the call to unlock or make sure to unlock before that return?